### ISSUE TITLE
Font rendering artifacts when calling DrawText() with a long multi-line string

### ISSUE THREAD

**Issue by AtwindYu** [SixLabors/ImageSharp#572]
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of ImageSharp
- [x] I have verified if the problem exist in both `DEBUG` and `RELEASE` mode
- [x] I have searched [open](https://github.com/SixLabors/ImageSharp/issues) and [closed](https://github.com/SixLabors/ImageSharp/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
If the texts are covered with the entire A4 paper(300dpi), the words on the bottom right becomes pasted.

### Steps to Reproduce
Key code:
```C#
TextOptions = new TextGraphicsOptions
            {
                Antialias = true,
                ApplyKerning = true, 
                VerticalAlignment = VerticalAlignment.Top,
                HorizontalAlignment = HorizontalAlignment.Left,
                WrapTextWidth = PageSize.Width
            };


 using (Image<Rgba32> img = new Image<Rgba32>(PageSize.Width, PageSize.Height))
            {
                img.MetaData.HorizontalResolution = 300;
                img.MetaData.VerticalResolution = 300;

                img.Mutate(x =>
                    x.Fill(Rgba32.White)
                        .DrawText(TextOptions, text, Font, Rgba32.Black, new PointF(10, 5))

                );
                img.Save($"output/0.png");            

```

### System Configuration

- ImageSharp version: SixLabors.ImageSharp.Drawing 1.0.0-beta0003
- Environment (Operating system, version and so on):win10 + vs2017
- .NET Framework version: .Net Core 2.0

**Comment by JimBobSquarePants** [comment]
Hi @AtwindYu 

Thanks for raising this issue. Would it be possible to add an image demonstrating the blurring? 

Cheers

James

**Comment by AtwindYu** [comment]
![01](https://user-images.githubusercontent.com/12068896/39958014-e6f16422-562e-11e8-9183-75826033b6fa.png)
![02](https://user-images.githubusercontent.com/12068896/39958015-e8db71c4-562e-11e8-9273-997fa65a6846.png)

**Comment by tocsoft** [comment]
I have a feeling that it could related to other issues we have seen around `Vector2` being retained as properties. The version of the Shapes package I've pointed to has just had them all removed.

**Comment by JimBobSquarePants** [comment]
Is that alph9 in the image a version number?

**Comment by AtwindYu** [comment]
@JimBobSquarePants How to set or change the alph9?

**Comment by JimBobSquarePants** [comment]
Are you connected to the right Myget feed for the nightlies? (It’s linked from the README) Alpha 9 is a very old version number from the old MyGet feed.

https://github.com/SixLabors/ImageSharp/blob/master/README.md#installation

**Comment by antonfirsov** [comment]
@JimBobSquarePants the example code suggests that he is not using an alpha.

@AtwindYu the question is the source of the marked information in the sample output image you posted in this thread. Is it an old output?

**Comment by AtwindYu** [comment]
Hi,There is my test code:
```C#
  [TestMethod]
        public void CreateA4300FullPageOutputBlurryTest()
        {
            var text = @"YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
YoushouldnotDisposetheoutputImageStreampassedtotheFileStreamResultobjectThatclassbecomestheownerandwillreleaseithemselvesYoushouldnotDisposethe
";

           var  textOptions = new TextGraphicsOptions
            {
                Antialias = true,
                ApplyKerning = true,
                VerticalAlignment = VerticalAlignment.Top,
                HorizontalAlignment = HorizontalAlignment.Left,
                //WrapTextWidth = PageSize.Width
            };
            var fo = SixLabors.Fonts.SystemFonts.Find("Microsoft Sans Serif");
            var font = new Font(fo, 36, FontStyle.Regular);
            using (Image<Rgba32> img = new Image<Rgba32>(2480, 3508))
            {
                img.MetaData.HorizontalResolution = 300;
                img.MetaData.VerticalResolution = 300;

                img.Mutate(x =>
                    x.Fill(Rgba32.White)
                        .DrawText(textOptions, text, font, Rgba32.Black, new PointF(10, 5))

                );

                img.Save("output/a4300.png");
                
            }

        }

```

Here is my nuget packages

![image](https://user-images.githubusercontent.com/12068896/39965878-2aece2dc-56d4-11e8-92c7-6fcb229964e2.png)

**Comment by JimBobSquarePants** [comment]
@AtwindYu That all make sense now. Thanks for the updated sample and detail. That’s really really helpful. :+1:

**Comment by JimBobSquarePants** [comment]
Issue confirmed using current latest of all packages from nightlies.

``` xml
    <PackageReference Include="SixLabors.Core" Version="1.0.0-ci0007" />
    <PackageReference Include="SixLabors.Fonts" Version="1.0.0-ci0022" />
    <PackageReference Include="SixLabors.ImageSharp" Version="1.0.0-dev001327" />
    <PackageReference Include="SixLabors.ImageSharp.Drawing" Version="1.0.0-dev001327" />
    <PackageReference Include="SixLabors.Shapes" Version="1.0.0-ci0018" />
    <PackageReference Include="SixLabors.Shapes.Text" Version="1.0.0-ci0018" />
```

![is-bug](https://user-images.githubusercontent.com/385879/39968349-bf35262c-570e-11e8-95aa-93e24a71419d.JPG)

**Comment by diegogarciaa** [comment]
Hi, I am having the same issue, is there any workaround ?

Thanks.

**Comment by antonfirsov** [comment]
@diegogarciaa try rendering smaller parts of the text on manually defined locations.

**Comment by woutware** [comment]
I've patched the issue: https://github.com/SixLabors/Shapes/pull/43.

Notice that the problem appears only for larger x, y coordinates (roughly > 1500), and it gets worse the larger the coordinates are (bottom right of bitmap is worst). This indicates that there's a floating point rounding problem: floating point is relatively precise around zero, but less precise the further from zero you get.

From this stand point a future improvement could be using the center of the bitmap as (0, 0), this will reduce the size of a a multiplication by a factor 4, improving the precision by the same factor.

Here's a quick program to reproduce the issue quicker:

```
        public static void CreateA4300FullPageOutputBlurryTest2() {

            var textOptions = new TextGraphicsOptions {
                Antialias = false,
                ApplyKerning = true,
                VerticalAlignment = VerticalAlignment.Top,
                HorizontalAlignment = HorizontalAlignment.Left,
                //WrapTextWidth = PageSize.Width
            };
            var fo = SixLabors.Fonts.SystemFonts.Find("Microsoft Sans Serif");
            var font = new Font(fo, 36, FontStyle.Regular);
            //int size = 2500;
            int size = 1500;
            using (Image<Rgba32> img = new Image<Rgba32>(size, size)) {
                img.MetaData.HorizontalResolution = 300;
                img.MetaData.VerticalResolution = 300;

                img.Mutate(x =>
                    x.Fill(Rgba32.White)
                        .DrawText(textOptions, "O", font, Rgba32.Black, new PointF(img.Width - 80, img.Height - 50))
                );

                img.Save(@"D:\support\image2.png");

            }
        }
```

Problem occurs at `y	>= 1478.5f`, set breakpoint condition `start.Y	>= 1478.5f` at `InternalPath.FindIntersections`. It will report 1 intersection for the outer polygon of the letter 'O' instead of 2.

**Comment by JimBobSquarePants** [comment]
Closing now as in nightlies

**Comment by antonfirsov** [comment]
Reopening this for now, see:
https://github.com/SixLabors/ImageSharp/issues/496#issuecomment-391323991

**Comment by antonfirsov** [comment]
Fixed with #596  in beta4.
